libraries/SITL/SIM_Sailboat.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+    Sailboat simulator class
+
+    see explanation of lift and drag explained here: https://en.wikipedia.org/wiki/Forces_on_sails
+
+    To-Do: add heel handling by calculating lateral force from wind vs gravity force from heel to arrive at roll rate or acceleration
+*/
+
+#include "SIM_Sailboat.h"
+#include <AP_Math/AP_Math.h>
+#include <string.h>
+#include <stdio.h>
+
+namespace SITL {
+
+Sailboat::Sailboat(const char *home_str, const char *frame_str) :
+    Aircraft(home_str, frame_str),
+    max_wheel_turn(35),
+    turning_circle(1.8)
+{
+}
+
+// calculate the lift and drag as values from 0 to 1
+// given an apparent wind speed in m/s and angle-of-attack in degrees
+void Sailboat::calc_lift_and_drag(float wind_speed, float angle_of_attack_deg, float& lift, float& drag)
+{
+    // check extremes
+    if (angle_of_attack_deg <= 0.0f) {
```

Comment:
I'm not a boat person, but in manned aviation it's quite possible to have a negative angle of attack, and I can't think of any reason why the boat can't have one? I'm guessing it's because the table you have doesn't map out correctly?